src/core/src/runtime/compute_hash.cpp

```diff
@@ -822,78 +826,90 @@ void ComputeHash<isa>::fold_to_64(const Vmm& v_dst) {
 
 size_t compute_hash(const void* src, size_t size) {
 #ifdef OV_CORE_USE_XBYAK_JIT
-    if (Generator::mayiuse(avx2)) {
-        uint64_t result = 0lu;
-
-        // Parallel section
-        constexpr uint64_t min_wa_per_thread = 131072lu;  // 2^17
-        const uint64_t size_u64 = static_cast<uint64_t>(size);
-        if (size_u64 >= min_wa_per_thread * 2lu) {
-            static auto first_thr_kernel = Generator::mayiuse(avx512_core)
-                                               ? jit::ComputeHash<avx512_core>::create({jit::FIRST_THREAD})
-                                               : jit::ComputeHash<avx2>::create({jit::FIRST_THREAD});
-            static auto n_thr_kernel = Generator::mayiuse(avx512_core)
-                                           ? jit::ComputeHash<avx512_core>::create({jit::N_THREAD})
-                                           : jit::ComputeHash<avx2>::create({jit::N_THREAD});
-            static auto final_fold_kernel = Generator::mayiuse(avx512_core)
-                                                ? jit::ComputeHash<avx512_core>::create({jit::FINAL_FOLD})
-                                                : jit::ComputeHash<avx2>::create({jit::FINAL_FOLD});
-
-            static const uint64_t max_thr_num = 2lu;
-            uint64_t thr_num = std::min(size_u64 / min_wa_per_thread, max_thr_num);
-            const uint64_t el_per_thread =
-                first_thr_kernel->get_vlen() * ((size_u64 / thr_num) / first_thr_kernel->get_vlen());
-            std::vector<uint8_t> intermediate(thr_num * first_thr_kernel->get_vlen());
-
-            parallel_nt_static(static_cast<int>(thr_num), [&](const int ithr, const int nthr) {
-                uint64_t start = el_per_thread * ithr;
-                if (start >= size_u64) {
-                    return;
-                }
-                uint64_t work_amount = (el_per_thread + start > size_u64) ? size_u64 - start : el_per_thread;
+
+#    if defined(_WIN32)
+    HANDLE handle = GetCurrentProcess();
+    PROCESS_MITIGATION_DYNAMIC_CODE_POLICY dynamic_code_policy = {0};
+    GetProcessMitigationPolicy(handle, ProcessDynamicCodePolicy, &dynamic_code_policy, sizeof(dynamic_code_policy));
+    if (dynamic_code_policy.ProhibitDynamicCode != TRUE) {
+#    endif
```

Comment:
Can some helper functio be added which returns bool flag.
For windows it do some action and return true/false for other OSes it can return constexpr value
and here will be just check e.g.:
``` if (may_uses_dynamic_code()){```

Reply:
Good point. Can we implement this in the master branch only as part of #29006?